iceoryx_hoofs/test/moduletests/test_posix_unnamed_semaphore.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/posix_wrapper/unnamed_semaphore.hpp"
+#include "test.hpp"
+
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::posix;
+using namespace iox::cxx;
+
+class UnnamedSemaphoreTest : public Test
+{
+  public:
+    void SetUp() override
+    {
+    }
+
+    void TearDown() override
+    {
+    }
+
+    optional<UnnamedSemaphore> sut;
+};
+
+TEST_F(UnnamedSemaphoreTest, DefaultInitialValueIsZero)
+{
+    ASSERT_FALSE(UnnamedSemaphoreBuilder().create(sut).has_error());
+    EXPECT_THAT(0U, sut->getState().expect("Failed to access semaphore").value);
+}
```

Comment:
Missing test ids ... and could you switch the parameter from `EXPECT_THAT` to have the actual value first and expected one second?